api/v1alpha1/release_types_test.go

```diff
@@ -0,0 +1,446 @@
+			|      Now()     | ConditionTrue    |     true     |  true    |  yes |
+			|--------------------------------------------------------------------|
+
+			When no NOOP:
+			* expect zero or nil CompletionTime to be set to Now().
+			* expect setStatusConditionWithMessage() to append a new condition.
+			* * expect condition.Status: ConditionFalse
+			* * expect condition.Reason: to match the reason arg
+			* * expect condition.Message: to match the message arg
+		*/
+		It("should NOOP when IsDone() is true, and Status.CompletionTime is not nil", func() {
+			args := commonArgs{
+				reason:  ReleaseReasonValidationError,
+				message: "cow say m00",
+			}
+			r.Status.Conditions[0] = metav1.Condition{
+				Type:    releaseConditionType,
+				Status:  metav1.ConditionTrue,
+				Reason:  ReleaseReasonTargetDisabledError.String(),
+				Message: "Testcase one message string",
+			}
+			r.MarkFailed(args.reason, args.message)
+			Expect(len(r.Status.Conditions)).To(Equal(1))
+			Expect(r.Status.Conditions[0]).To(MatchFields(IgnoreMissing|IgnoreExtras, Fields{
+				"Status":  Equal(metav1.ConditionTrue),
+				"Type":    Equal(releaseConditionType),
+				"Reason":  Equal(ReleaseReasonTargetDisabledError.String()),
+				"Message": Equal("Testcase one message string"),
+			}))
+		})
+
+		It("should not NOOP when CompletionTime is nil, and IsDone returns false", func() {
+			args := commonArgs{
+				reason:  ReleaseReasonValidationError,
+				message: "what does the fox say",
+			}
+			r.Status.CompletionTime = nil
+			r.Status.Conditions[0] = metav1.Condition{
+				Type:    releaseConditionType,
+				Status:  metav1.ConditionUnknown,
+				Reason:  ReleaseReasonTargetDisabledError.String(),
+				Message: "Testcase two message string",
+			}
+			r.MarkFailed(args.reason, args.message)
+			Expect(len(r.Status.Conditions)).To(Equal(1))
+			Expect(r.Status.Conditions[0]).To(MatchFields(IgnoreMissing|IgnoreExtras, Fields{
+				"Status":  Equal(metav1.ConditionFalse),
+				"Type":    Equal(releaseConditionType),
+				"Reason":  Equal(ReleaseReasonValidationError.String()),
+				"Message": Equal("what does the fox say"),
+			}))
+		})
+	})
+
+	Context("When MarkInvalid method is called", func() {
+		/*
+			|------------------------------------|
+			|            NOOP matrix             |
+			|------------------------------------|
+			| condition.Status | IsDone() | NOOP |
+			|------------------|----------|------|
+			|       nil        |  false   |  no  |
+			| ConditionUnknown |  false   |  no  |
+			| ConditionTrue    |  true    |  yes |
+			|------------------------------------|
+
+			When no NOOP:
+			* expect setStatusConditionWithMessage() to append a new condition.
+			* * expect condition.Status: ConditionFalse
+			* * expect condition.Reason: to match the reason arg
+			* * expect condition.Message: to match the message arg
+		*/
+		It("should not NOOP when IsDone() returns false", func() {
```

Comment:
No need for a table. Let's fix this in the same way than the previous ones:
```suggestion
		It("should register an Invalid status if the Release hasn't finished", func() {
```